File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/FindCredential.java
package org.cloudfoundry.credhub.audit.entities;

import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class FindCredential implements RequestDetails {
  private String nameLike;
  private String path;
  private Boolean paths;

  private String expiresWithinDays;

  public String getExpiresWithinDays() {
    return expiresWithinDays;
  }

  public void setExpiresWithinDays(final String expiresWithinDays) {
    this.expiresWithinDays = expiresWithinDays;
  }

  public String getNameLike() {
    return nameLike;
  }

  public void setNameLike(final String nameLike) {
    this.nameLike = nameLike;
  }

  public String getPath() {
    return path;
  }

  public void setPath(final String path) {
    this.path = path;
  }

  public Boolean getPaths() {
    return paths;
  }

  public void setPaths(final Boolean paths) {
    this.paths = paths;
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.FIND;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/FindCredential.kt
package org.cloudfoundry.credhub.audit.entities

import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails

class FindCredential : RequestDetails {
    var nameLike: String? = null
    var path: String? = null
    var paths: Boolean? = null

    var expiresWithinDays: String? = null

    override fun operation(): OperationDeviceAction = OperationDeviceAction.FIND
}
